public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx

```diff
@@ -13,100 +13,80 @@ interface Props {
 }
 
 interface State {
-  searchHits: DashboardSearchHit[];
+  resolvedLinks: ResolvedLinkDTO[];
 }
 
 export class DashboardLinksDashboard extends PureComponent<Props, State> {
-  state = { searchHits: [] as DashboardSearchHit[] };
-  componentDidMount() {
-    if (!this.props.link.asDropdown) {
-      this.onDropDownClick();
+  state: State = { resolvedLinks: [] };
+
+  componentDidUpdate(prevProps: Readonly<Props>) {
+    if (!this.props.link.asDropdown && prevProps.linkInfo !== this.props.linkInfo) {
+      this.onResolveLinks();
     }
   }
 
-  onDropDownClick = () => {
+  onResolveLinks = async () => {
     const { dashboardId, link } = this.props;
 
-    const limit = 7;
-    getBackendSrv()
-      .search({ tag: link.tags, limit })
-      .then((dashboards: DashboardSearchHit[]) => {
-        const processed = dashboards
-          .filter(dash => dash.id !== dashboardId)
-          .map(dash => {
-            return {
-              ...dash,
-              url: getLinkSrv().getLinkUrl(dash),
-            };
-          });
-
-        this.setState({
-          searchHits: processed,
-        });
-      });
+    const searchHits = await searchForTags(link);
```

Comment:
The limit = 7 was suuuuper weird. Is this a mistake or was it like this before migration? :D

Reply:
A mistake while migrating to React it seems.